core/src/avm2/activation.rs

```diff
@@ -637,7 +637,7 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     }
 
     fn op_push_short(&mut self, value: u32) -> Result<FrameControl<'gc>, Error> {
-        self.context.avm2.push(value);
+        self.context.avm2.push(value as i32);
```

Comment:
Should this cast be moved to `swf`, `Op::PushShort { value: i32 }`? Probably so if it's misdocumented.

Reply:
`swf` actually has a TODO on this opcode asking if this is really a u30. The AVM2 documentation from Adobe claims it's *stored* as a `u30`, which is technically accurate, but also unhelpful.

I peeked at JPEXS's source and it appears to parse this by reading a `u30` from the instruction stream and then casting it to a Java `short`, equivalent to a Rust `i16`. That also works for my tests, so I'll adjust `swf` as such.